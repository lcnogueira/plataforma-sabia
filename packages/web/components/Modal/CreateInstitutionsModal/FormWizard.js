import React from 'react';
import PropTypes from 'prop-types';
import { Form, Actions } from '../../Form';
import { Button } from './styles';

const getForm = (steps, currentStep) => {
	const findStep = steps.find((step) => step.slug === currentStep);
	return currentStep !== '' && findStep ? findStep.form : steps[0].form;
};

const FormWizard = ({
	steps,
	currentStep,
	onSubmit,
	onPrev,
	submitting,
	closeModal,
	defaultValues,
}) => {
	const CurrentFormStep = getForm(steps, currentStep);
	const currentStepSlug = currentStep || steps[0].slug;

	let currentStepIndex = 0;
	steps.forEach((step, i) => {
		if (step.slug === currentStepSlug) {
			currentStepIndex = i;
		}
	});

	const nextStep =
		currentStepIndex === steps.length - 1 ? false : steps[currentStepIndex + 1].slug;
	const prevStep = currentStepIndex === 0 ? false : steps[currentStepIndex - 1].slug;
	const lastStep = currentStepIndex === steps.length - 1;

	/**
	 * Handles going back in the form wizard.
	 */
	const handlePrev = () => {
		onPrev({ prevStep });
	};

	/**
	 * Handles submitting the form data for each step of the form wizard.
	 *
	 * @param {object} formData An object containing all the form data.
	 * @param {object} form A instance of the `useForm` hook.
	 */
	const handleSubmit = (formData, form) => {
		onSubmit({ data: formData, nextStep }, form);
	};

	return (
		<Form onSubmit={handleSubmit} defaultValues={defaultValues}>
			{CurrentFormStep && <CurrentFormStep />}

			<Actions center>
				{prevStep ? (
					<Button
						type="button"
						variant="outlined"
						disabed={submitting}
						onClick={handlePrev}
					>
						Voltar
					</Button>
				) : (
					<Button
						type="button"
						variant="outlined"
						onClick={() => closeModal()}
						disabled={submitting}
					>
						Cancelar
					</Button>
				)}
				{nextStep && (
					<Button disabled={submitting} type="submit">
						{submitting ? 'Salvando...' : 'Continuar'}
					</Button>
				)}
				{lastStep && (
					<Button disabled={submitting} type="submit">
						Concluir
					</Button>
				)}
			</Actions>
		</Form>
	);
};

FormWizard.propTypes = {
	onSubmit: PropTypes.func,
	onPrev: PropTypes.func,
	submitting: PropTypes.bool,
	steps: PropTypes.arrayOf(
		PropTypes.shape({
			slug: PropTypes.string.isRequired,
			form: PropTypes.elementType,
		}),
	).isRequired,
	currentStep: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]).isRequired,
	closeModal: PropTypes.func.isRequired,
	defaultValues: PropTypes.shape({}),
};

FormWizard.defaultProps = {
	submitting: false,
	onSubmit: () => {},
	onPrev: () => {},
	defaultValues: {},
};

export default FormWizard;
