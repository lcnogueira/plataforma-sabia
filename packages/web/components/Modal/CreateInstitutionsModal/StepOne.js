/* eslint-disable react/jsx-props-no-spreading */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import PlacesAutocomplete, { geocodeByPlaceId } from 'react-places-autocomplete';
import {
	InputField,
	MaskedInputField,
	RequiredIndicator,
	SelectField,
	InputHiddenField,
} from '../../Form';
import { Cell, Row } from '../../Common';
import { InputLabel } from '../../Form/styles';
import { Suggestion, GoogleAddressSugestions } from './styles';
import { STATES } from '../../../utils/enums/states.enum';
import { mapArrayOfObjectToSelect } from '../../../utils/helper';

const StepOne = ({ form }) => {
	const { setValue } = form;
	const [loadingPlace, setLoadingPlace] = useState(false);
	const [addressFields, setAddressFields] = useState(false);
	const [addressInput, setAddressInput] = useState('');

	const onSelect = async (placeId) => {
		setLoadingPlace(true);
		setAddressFields(true);
		const response = await geocodeByPlaceId(placeId);
		if (response) {
			const address = response[0];
			const state = address.address_components[4]?.short_name;
			setValue('address', address.address_components[1]?.long_name);
			setValue('district', address.address_components[2]?.long_name);
			setValue('zipcode', address.address_components[6]?.long_name);
			setValue('state', state);
			setValue('city', address.address_components[3]?.long_name);
			setValue('lat', address.geometry.location.lat());
			setValue('lng', address.geometry.location.lng());
			setLoadingPlace(false);
		}
	};

	return (
		<>
			<Row>
				<Cell col={12}>
					<InputField
						form={form}
						label="Nome"
						name="name"
						type="text"
						validation={{ required: true }}
						variant="gray"
					/>
				</Cell>
			</Row>
			<Row>
				<Cell col={4}>
					<InputField
						form={form}
						label="Sigla"
						name="initials"
						type="text"
						validation={{ required: true }}
						variant="gray"
					/>
				</Cell>
				<Cell col={8}>
					<MaskedInputField
						form={form}
						name="cnpj"
						label="CNPJ"
						placeholder=""
						mask="99.999.999/9999-99"
						pattern={/^\d{2}\.\d{3}\.\d{3}\/\d{4}\-\d{2}$/} // eslint-disable-line no-useless-escape
						validation={{ required: true }}
						variant="gray"
					/>
				</Cell>
			</Row>
			<Row>
				<Cell col={12}>
					<InputLabel>
						Logradouro <RequiredIndicator />
					</InputLabel>
					<PlacesAutocomplete
						id="placeId"
						name="place_id"
						variant="gray"
						value={addressInput}
						onChange={(value) => setAddressInput(value)}
						onSelect={(value, placeId) => onSelect(placeId)}
					>
						{({
							getInputProps,
							suggestions,
							getSuggestionItemProps,
							loading: placesLoading,
						}) => (
							<div>
								<InputField
									{...getInputProps({
										placeholder: 'Procurar localidades...',
										className: 'location-search-input',
									})}
									form={{ register: () => {} }}
									variant="gray"
									name="places"
								/>
								<div className="autocomplete-dropdown-container">
									{placesLoading && (
										<GoogleAddressSugestions>
											Carregando...
										</GoogleAddressSugestions>
									)}
									<GoogleAddressSugestions>
										{suggestions.map((suggestion) => {
											const style = suggestion.active
												? { backgroundColor: '#fafafa' }
												: { backgroundColor: '#fff' };
											return (
												<div
													{...getSuggestionItemProps(suggestion, {
														style,
													})}
													key={suggestion.placeId}
												>
													<Suggestion>
														{suggestion.description}
													</Suggestion>
												</div>
											);
										})}
									</GoogleAddressSugestions>
								</div>
							</div>
						)}
					</PlacesAutocomplete>
				</Cell>
			</Row>
			{addressFields && addressInput !== '' && (
				<>
					<Row>
						<Cell col={8}>
							<InputField
								form={form}
								label="Bairro"
								name="district"
								type="text"
								validation={{ required: true }}
								variant="gray"
								disabled={loadingPlace}
							/>
						</Cell>
						<Cell col={4}>
							<MaskedInputField
								form={form}
								name="zipcode"
								label="CEP"
								placeholder="12345-123"
								mask="99999-999"
								pattern={/^\d{5}-\d{3}$/}
								validation={{ required: true }}
								variant="gray"
								disabled={loadingPlace}
							/>
						</Cell>
					</Row>
					<Row>
						<Cell col={6}>
							<SelectField
								form={form}
								name="state"
								label="UF"
								validation={{ required: true }}
								onChange={([option]) => {
									setValue('state', option.value);
									return option;
								}}
								options={mapArrayOfObjectToSelect(STATES, 'initials', 'initials')}
								variant="gray"
								disabled={loadingPlace}
							/>
						</Cell>
						<Cell col={6}>
							<InputField
								form={form}
								label="Cidade"
								name="city"
								type="text"
								validation={{ required: true }}
								variant="gray"
								disabled={loadingPlace}
							/>
						</Cell>
					</Row>
				</>
			)}
			<InputHiddenField form={form} name="address" />
			<InputHiddenField form={form} name="lat" />
			<InputHiddenField form={form} name="lng" />
		</>
	);
};

StepOne.propTypes = {
	form: PropTypes.shape({
		watch: PropTypes.func,
		getValues: PropTypes.func,
		setValue: PropTypes.func,
	}),
};

StepOne.defaultProps = {
	form: {},
};

export default StepOne;
