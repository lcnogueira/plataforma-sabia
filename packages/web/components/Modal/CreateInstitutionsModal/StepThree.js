/* eslint-disable react/jsx-props-no-spreading */
import React, { useRef, useState } from 'react';
import PropTypes from 'prop-types';
import Dropzone from 'react-dropzone';
import { InputHiddenField } from '../../Form';
import { Cell, Row } from '../../Common';
import { InputLabel } from '../../Form/styles';
import { upload } from '../../../services/uploads';

import * as S from './styles';

const StepThree = ({ form }) => {
	const [logoPreview, setLogoPreview] = useState(null);
	const [uploadError, setUploadError] = useState(null);
	const logoRef = useRef(null);
	const { setValue } = form;

	// eslint-disable-next-line consistent-return
	const onDropAttachment = async (acceptedFiles) => {
		if (!acceptedFiles) return null;

		const formData = new FormData();
		if (acceptedFiles.length !== 0) {
			formData.append(`files[0]`, acceptedFiles[0], acceptedFiles[0].name);
		}

		const response = await upload(formData);

		if (response.status === 200) {
			const { id, url } = response.data[0];
			setLogoPreview(url);
			setValue('logo_id', id);
		} else {
			setUploadError(response.data.error.message[0].message);
		}
	};

	return (
		<>
			<Row>
				<Cell col={12}>
					<S.DropzoneWrapper>
						<InputLabel>Logo</InputLabel>
						<Dropzone
							accept="image/*"
							onDrop={(acceptedFiles) => onDropAttachment(acceptedFiles)}
							ref={logoRef}
							multiple={false}
						>
							{({ getRootProps, getInputProps }) => (
								<S.LogoDropzone {...getRootProps()}>
									<input name="logo" {...getInputProps()} />
									{logoPreview && (
										<S.LogoPreview url={logoPreview} alt="Prévia da logo" />
									)}
									<S.PreviewHelp>
										Clique ou arraste a logo da organização aqui.
									</S.PreviewHelp>
									{uploadError && <S.UploadError>{uploadError}</S.UploadError>}
								</S.LogoDropzone>
							)}
						</Dropzone>
					</S.DropzoneWrapper>
				</Cell>
			</Row>
			<InputHiddenField form={form} name="logo_id" />
		</>
	);
};

StepThree.propTypes = {
	form: PropTypes.shape({
		watch: PropTypes.func,
		getValues: PropTypes.func,
		setValue: PropTypes.func,
	}),
};

StepThree.defaultProps = {
	form: {},
};

export default StepThree;
