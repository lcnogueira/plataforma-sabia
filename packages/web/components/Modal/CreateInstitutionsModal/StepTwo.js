/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import PropTypes from 'prop-types';
import { InputField, MaskedInputField, SelectField } from '../../Form';
import { Cell, Row } from '../../Common';
import { beforeMaskedValueChange } from '../../../utils/helper';

const typeOptions = [
	{
		label: 'Público',
		value: 'public',
	},
	{
		label: 'Privado',
		value: 'private',
	},
	{
		label: 'Misto',
		value: 'mixed',
	},
	{
		label: 'Outro',
		value: 'other',
	},
];

const categoryOptions = [
	{
		label: 'Universidade',
		value: 'university',
	},
	{
		label: 'Instituto',
		value: 'institute',
	},
	{
		label: 'Associação',
		value: 'association',
	},
	{
		label: 'Fundação',
		value: 'foundation',
	},
	{
		label: 'Cooperativa',
		value: 'cooperative',
	},
	{
		label: 'Companhia',
		value: 'company',
	},
	{
		label: 'Outro',
		value: 'other',
	},
];

const StepTwo = ({ form }) => {
	const { setValue } = form;

	return (
		<>
			<Row>
				<Cell col={12}>
					<InputField
						form={form}
						label="E-mail para contato"
						name="email"
						type="email"
						variant="gray"
					/>
				</Cell>
			</Row>
			<Row>
				<Cell col={6}>
					<MaskedInputField
						form={form}
						name="phone_number"
						alwaysShowMask={false}
						label="Telefone"
						mask="(99) 9999-99999"
						maskChar={null}
						beforeMaskedValueChange={beforeMaskedValueChange}
						pattern={/(\(?\d{2}\)?\s)?(\d{4,5}-\d{4})/}
						variant="gray"
						defaultValue=""
					/>
				</Cell>
				<Cell col={6}>
					<InputField
						form={form}
						label="Website"
						name="website"
						placeholder="https://exemplo.org"
						type="text"
						variant="gray"
					/>
				</Cell>
			</Row>
			<Row>
				<Cell col={6}>
					<SelectField
						form={form}
						name="type"
						label="Tipo"
						validation={{ required: true }}
						onChange={([option]) => {
							setValue('type', option.value);
							return option;
						}}
						options={typeOptions}
						variant="gray"
					/>
				</Cell>
				<Cell col={6}>
					<SelectField
						form={form}
						name="category"
						label="Categoria"
						validation={{ required: true }}
						onChange={([option]) => {
							setValue('category', option.value);
							return option;
						}}
						options={categoryOptions}
						variant="gray"
					/>
				</Cell>
			</Row>
		</>
	);
};

StepTwo.propTypes = {
	form: PropTypes.shape({
		watch: PropTypes.func,
		getValues: PropTypes.func,
		setValue: PropTypes.func,
	}),
};

StepTwo.defaultProps = {
	form: {},
};

export default StepTwo;
