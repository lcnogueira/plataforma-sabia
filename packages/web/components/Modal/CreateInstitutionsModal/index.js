import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Form, Actions, InputField, MaskedInputField, SelectField } from '../../Form';
import { Container, Button } from './styles';
import { Cell, Row } from '../../Common';

const InstitutionsForm = ({ form, closeModal, loading }) => (
	<>
		<Row>
			<Cell col={12}>
				<InputField
					form={form}
					label="Nome"
					name="name"
					type="text"
					validation={{ required: true }}
					variant="gray"
				/>
			</Cell>
		</Row>
		<Row>
			<Cell col={4}>
				<InputField
					form={form}
					label="Sigla"
					name="initials"
					type="text"
					validation={{ required: true }}
					variant="gray"
				/>
			</Cell>
			<Cell col={8}>
				<MaskedInputField
					form={form}
					name="cnpj"
					label="CNPJ"
					placeholder=""
					mask="999.999.999-99"
					mask="99.999.999/9999-99"
					pattern={/^\d{2}\.\d{3}\.\d{3}\/\d{4}\-\d{2}$/}
					validation={{ required: true }}
					variant="gray"
				/>
			</Cell>
		</Row>
		<Row>
			<Cell col={8}>
				<InputField
					form={form}
					label="Logradouro"
					name="address"
					type="text"
					validation={{ required: true }}
					variant="gray"
				/>
			</Cell>
			<Cell col={4}>
				<MaskedInputField
					form={form}
					name="zipcode"
					label="CEP"
					placeholder="12345-123"
					mask="99999-999"
					pattern={/^\d{5}-\d{3}$/}
					validation={{ required: true }}
					variant="gray"
				/>
			</Cell>
		</Row>
		<Row>
			<Cell col={6}>
				<SelectField
					form={form}
					name="uf"
					label="UF"
					validation={{ required: true }}
					variant="gray"
				/>
			</Cell>
			<Cell col={6}>
				<SelectField
					form={form}
					name="city"
					label="Cidade"
					validation={{ required: true }}
					variant="gray"
				/>
			</Cell>
		</Row>
		<Actions center>
			<Button
				type="button"
				variant="outlined"
				onClick={() => closeModal()}
				disabled={loading}
			>
				Cancelar
			</Button>
			<Button type="submit" disabled={loading}>
				Cadastrar
			</Button>
		</Actions>
	</>
);

const CreateInstitutionsModal = ({ closeModal }) => {
const [isSubmitting, setIsSubmitting] = useState(false);

const handleSubmit = async () => {};

return (
	<Container>
		<h3>Cadastrar nova Organização</h3>
		<Form onSubmit={handleSubmit} aria-label="form">
			<InstitutionsForm loading={isSubmitting} closeModal={closeModal} />
		</Form>
	</Container>
);
};

CreateInstitutionsModal.propTypes = {
	closeModal: PropTypes.func.isRequired,
};

export default CreateInstitutionsModal;
