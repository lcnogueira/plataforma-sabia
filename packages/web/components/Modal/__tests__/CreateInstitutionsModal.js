import React from 'react';

import { render, screen } from 'test-utils';
import CreateInstitutionsModal from '../CreateInstitutionsModal';

const closeModalMock = jest.fn();

describe('<CreateInstitutionsModal />', () => {
	it('should render correctly', () => {
		const { container } = render(<CreateInstitutionsModal closeModal={closeModalMock} />);

		expect(screen.getByRole('form')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
