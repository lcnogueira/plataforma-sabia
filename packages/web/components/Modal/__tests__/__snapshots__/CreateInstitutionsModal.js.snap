// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateInstitutionsModal /> should render correctly 1`] = `
.c5 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c8 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  opacity: 1;
  border: 1px solid hsl(0,0%,93%);
  border-radius: 0.5rem;
  background: hsl(0,0%,93%);
}

.c8::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c8::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c8:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c8::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 {
  font-size: 1.4rem;
  line-height: 14px;
}

.c11 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  color: hsla(4,85%,57%);
  margin-left: 0.5rem;
  display: inline-block;
}

.c14 {
  width: 100%;
  margin: 0.5rem 0;
  font-size: 1.4rem;
}

.c12 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: hsla(0,0%,100%);
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
  opacity: 1;
  border: 1px solid hsl(0,0%,93%);
  border-radius: 0.5rem;
  background: hsl(0,0%,93%);
}

.c12::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c12::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c12:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c12::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c1 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c1 button,
.c1 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c1 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c15 > a[href] {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 0 1rem;
}

.c15 > button {
  margin-top: 0.5rem;
  margin-left: 1.5rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-top: 0rem;
  margin-bottom: 1rem;
  background-color: transparent;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  -webkit-flex: 12;
  -ms-flex: 12;
  flex: 12;
  margin: 0 1rem 0 0;
  text-align: left;
}

.c9 {
  -webkit-flex: 4;
  -ms-flex: 4;
  flex: 4;
  margin: 0 1rem 0 0;
  text-align: left;
}

.c10 {
  -webkit-flex: 8;
  -ms-flex: 8;
  flex: 8;
  margin: 0 1rem 0 0;
  text-align: left;
}

.c13 {
  -webkit-flex: 6;
  -ms-flex: 6;
  flex: 6;
  margin: 0 1rem 0 0;
  text-align: left;
}

.c0 {
  background-color: hsla(0,0%,100%);
  border-radius: 5px;
  width: 100%;
  max-width: 51.2rem;
  padding: 0.5rem 1.8rem;
}

.c0 label {
  font-size: 1.6rem;
}

.c0 button {
  margin-top: 2rem;
}

.c0 h3 {
  font-size: 28px;
  font-style: normal;
  font-weight: 500;
  line-height: 33px;
  text-align: center;
  color: hsl(166,100%,34%);
  margin-top: 2.8rem;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  border: none;
  outline: none;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 1.4rem;
  line-height: 2.4rem;
  background: none;
  color: hsla(4,85%,57%);
  border: 2px solid hsla(4,85%,57%);
  padding: 0.2rem 0.6rem;
}

.c16 > svg {
  margin-right: 0.4rem;
}

.c16:disabled {
  pointer-events: none;
  opacity: 0.5;
}

.c16:hover,
.c16:focus {
  color: hsla(0,0%,100%);
  background-color: hsla(4,85%,57%);
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  border: none;
  outline: none;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 1.4rem;
  line-height: 2.4rem;
  background: hsl(166,100%,34%);
  color: hsla(0,0%,100%);
  padding: 0.4rem 0.8rem;
}

.c17 > svg {
  margin-right: 0.4rem;
}

.c17:disabled {
  pointer-events: none;
  opacity: 0.5;
}

.c17:hover,
.c17:focus {
  background: hsla(146,64%,36%);
}

@media (max-width:768px) {
  .c15 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }

  .c15 > button {
    margin-left: 0;
  }
}

@media (max-width:1170px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media (max-width:1170px) {
  .c3 {
    max-width: initial;
    margin: 0 0 1rem 0;
  }
}

@media (max-width:768px) {
  .c3 {
    margin: 0;
  }
}

@media (max-width:1170px) {
  .c9 {
    max-width: initial;
    margin: 0 0 1rem 0;
  }
}

@media (max-width:768px) {
  .c9 {
    margin: 0;
  }
}

@media (max-width:1170px) {
  .c10 {
    max-width: initial;
    margin: 0 0 1rem 0;
  }
}

@media (max-width:768px) {
  .c10 {
    margin: 0;
  }
}

@media (max-width:1170px) {
  .c13 {
    max-width: initial;
    margin: 0 0 1rem 0;
  }
}

@media (max-width:768px) {
  .c13 {
    margin: 0;
  }
}

<div>
  <section
    class="c0"
  >
    <h3>
      Cadastrar nova Organização
    </h3>
    <form
      aria-label="form"
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4"
          >
            <label
              class="c5"
              for="name"
            >
              Nome
               
              <span
                aria-label="Indicador visual de campo obrigatório"
                class="c6"
              >
                *
              </span>
            </label>
            <div
              class="c7"
            >
              <input
                aria-label="Nome"
                aria-required="true"
                class="c8"
                id="name"
                name="name"
                placeholder=""
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c9"
        >
          <div
            class="c4"
          >
            <label
              class="c5"
              for="initials"
            >
              Sigla
               
              <span
                aria-label="Indicador visual de campo obrigatório"
                class="c6"
              >
                *
              </span>
            </label>
            <div
              class="c7"
            >
              <input
                aria-label="Sigla"
                aria-required="true"
                class="c8"
                id="initials"
                name="initials"
                placeholder=""
                type="text"
              />
            </div>
          </div>
        </div>
        <div
          class="c10"
        >
          <div
            class="c11"
          >
            <label
              class="c5"
              for="cnpj"
            >
              CNPJ
               
              <span
                aria-label="Indicador visual de campo obrigatório"
                class="c6"
              >
                *
              </span>
            </label>
            <div
              class="c7"
            >
              <input
                aria-label="CNPJ"
                aria-required="true"
                class="c12"
                id="cnpj"
                name="cnpj"
                placeholder=""
                type="text"
                value=""
                variant="gray"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c10"
        >
          <div
            class="c4"
          >
            <label
              class="c5"
              for="address"
            >
              Logradouro
               
              <span
                aria-label="Indicador visual de campo obrigatório"
                class="c6"
              >
                *
              </span>
            </label>
            <div
              class="c7"
            >
              <input
                aria-label="Logradouro"
                aria-required="true"
                class="c8"
                id="address"
                name="address"
                placeholder=""
                type="text"
              />
            </div>
          </div>
        </div>
        <div
          class="c9"
        >
          <div
            class="c11"
          >
            <label
              class="c5"
              for="zipcode"
            >
              CEP
               
              <span
                aria-label="Indicador visual de campo obrigatório"
                class="c6"
              >
                *
              </span>
            </label>
            <div
              class="c7"
            >
              <input
                aria-label="CEP"
                aria-required="true"
                class="c12"
                id="zipcode"
                name="zipcode"
                placeholder="12345-123"
                type="text"
                value=""
                variant="gray"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c13"
        >
          <div
            class="c11"
          >
            <label
              class="c5"
              for="uf"
            >
              UF
              <span
                aria-label="Indicador visual de campo obrigatório"
                class="c6"
              >
                *
              </span>
            </label>
            <div
              class="c7"
            >
              <div
                class="c14 react-select-container css-2b097c-container"
                id="uf"
              >
                <div
                  class="react-select__control css-idjk6g-control"
                >
                  <div
                    class="react-select__value-container css-g1d714-ValueContainer"
                  >
                    <div
                      class="react-select__placeholder css-1wa3eu0-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      class="css-b8ldur-Input"
                    >
                      <div
                        class="react-select__input"
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-label="UF"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-2-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: 1.4rem; font-weight: 400; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class="react-select__indicator-separator css-1xjgjl1-IndicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="react-select__indicator react-select__dropdown-indicator css-xp98c8-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-6q0nyr-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <input
                  name="uf"
                  type="hidden"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="c13"
        >
          <div
            class="c11"
          >
            <label
              class="c5"
              for="city"
            >
              Cidade
              <span
                aria-label="Indicador visual de campo obrigatório"
                class="c6"
              >
                *
              </span>
            </label>
            <div
              class="c7"
            >
              <div
                class="c14 react-select-container css-2b097c-container"
                id="city"
              >
                <div
                  class="react-select__control css-idjk6g-control"
                >
                  <div
                    class="react-select__value-container css-g1d714-ValueContainer"
                  >
                    <div
                      class="react-select__placeholder css-1wa3eu0-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      class="css-b8ldur-Input"
                    >
                      <div
                        class="react-select__input"
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-label="Cidade"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-3-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: 1.4rem; font-weight: 400; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class="react-select__indicator-separator css-1xjgjl1-IndicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="react-select__indicator react-select__dropdown-indicator css-xp98c8-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-6q0nyr-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <input
                  name="city"
                  type="hidden"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c15"
      >
        <button
          class="c16"
          type="button"
        >
          Cancelar
        </button>
        <button
          class="c17"
          type="submit"
        >
          Cadastrar
        </button>
      </div>
    </form>
  </section>
</div>
`;
